lib/segment/src/index/sparse_index/sparse_index_config.rs

```diff
@@ -39,8 +40,25 @@ impl SparseIndexType {
     }
 }
 
+/// Storage types for vectors
+#[derive(Debug, Deserialize, Serialize, JsonSchema, Copy, Clone, PartialEq, Eq, Default)]
+pub enum SparseVectorIndexDatatype {
+    // Single-precision floating point
+    #[default]
+    Float32,
+    // Half-precision floating point
+    Float16,
+}
```

Comment:
Do we want [`#[serde(rename_all = "snake_case")]`](https://serde.rs/container-attrs.html#rename_all) here?